### ISSUE TITLE
Regression fixture generator and test suite

### ISSUE THREAD

**Issue by jobu711** [jobu711/options_arena#663]
# Task: Regression fixture generator and test suite

## Description
Build a fixture generator that queries historical wrong recommendations (high confidence + negative P&L) from the outcomes database and serializes them as JSON test fixtures. Then build a parametrized pytest suite that verifies current recommendations don't repeat known failures. This catches the "AI carries the same assumptions" blind spot.

## Acceptance Criteria
- [ ] `tools/generate_regression_fixtures.py` script queries wrong recommendations
- [ ] Fixtures include serialized `MarketContext` + `TickerScore` snapshots
- [ ] Fixtures stored as JSON in `tests/regression/fixtures/` (git-tracked)
- [ ] Parametrized pytest suite in `tests/regression/test_recommendation_regression.py`
- [ ] Tests assert new recommendations don't repeat high-confidence failures
- [ ] Tests use `TestModel` — no real API calls
- [ ] `@pytest.mark.regression` marker for selective running

## Technical Details
- New script: `tools/generate_regression_fixtures.py`
- New directory: `tests/regression/` with `conftest.py` and test file
- New directory: `tests/regression/fixtures/` for JSON fixture files
- Fixture format: `{ticker, market_context_json, ticker_score_json, actual_pnl_pct, original_direction, original_confidence}`
- Query: outcomes where `confidence >= 0.6 AND contract_return_pct < -20.0`
- Test assertion: if original was wrong with high confidence, new recommendation should be more cautious (lower confidence or different direction)

### Read First
Files that MUST be read before starting this task:
- `src/options_arena/data/_analytics.py` — outcome query patterns
- `src/options_arena/models/analytics.py` — ContractOutcome model
- `src/options_arena/models/analysis.py` — MarketContext, TickerScore models
- `tests/CLAUDE.md` — test conventions, markers, mocking rules

### Action
1. Create `tools/generate_regression_fixtures.py` script
   - WHY: generates fixtures from historical data, run manually when outcome data available
2. Query outcomes where confidence >= 0.6 AND contract_return_pct < -20.0
   - WHY: high confidence + large loss = known failure pattern
3. Serialize MarketContext + TickerScore as JSON fixture per ticker
   - WHY: reproducible test input, git-tracked
4. Create `tests/regression/conftest.py` with fixture loader
   - WHY: parametrize tests from fixture directory
5. Create `tests/regression/test_recommendation_regression.py` with parametrized tests
   - WHY: verify prompt changes don't repeat known failures
6. Add `@pytest.mark.regression` marker to all tests
   - WHY: selective running (`pytest -m regression`)
7. Register `regression` marker in `pytest.ini` or `pyproject.toml`
   - WHY: avoid PytestUnknownMarkWarning

### Verify
```
uv run pytest tests/regression/ -v -m regression
uv run ruff check tools/generate_regression_fixtures.py
uv run mypy tools/generate_regression_fixtures.py --strict
```

### Done
- Generator script runs against test database and produces fixtures
- At least 1 sample fixture committed (can be synthetic for bootstrap)
- Parametrized tests load fixtures and run assertions
- Tests marked with `@pytest.mark.regression`

### Test Files
- `tests/regression/test_recommendation_regression.py`

### Test Cases
```python
@pytest.mark.regression
class TestRecommendationRegression:
    @pytest.mark.parametrize("fixture", load_regression_fixtures())
    async def test_no_high_confidence_repeat(self, fixture) -> None:
        """Verify current model doesn't repeat known high-confidence failure."""

    def test_fixture_loader_handles_empty_dir(self) -> None:
        """Verify empty fixtures directory produces no test cases (not error)."""

    def test_fixture_format_valid(self) -> None:
        """Verify each fixture deserializes to valid MarketContext + TickerScore."""

    async def test_synthesis_reduces_confidence_on_failure(self) -> None:
        """Verify synthesis shows reduced confidence for known failure ticker."""

    def test_sample_fixture_exists(self) -> None:
        """Verify at least one fixture committed for bootstrap."""
```

### Edge Cases
- No fixture files — test suite collects 0 tests (not error)
- Fixture with missing fields — skip with warning
- Very old fixture (stale MarketContext) — still valid for regression check
- Generator finds no wrong recommendations — produces empty fixtures dir

### Mocking Strategy
- `TestModel` for desk/synthesis agent runs
- `ALLOW_MODEL_REQUESTS = False` at module level
- Mock Repository for generator script tests
- Synthetic fixture for bootstrap (committed to repo)

## Dependencies
- [ ] Task 002: Eval persistence (for querying outcome data)

## Effort Estimate
- Size: M
- Hours: 3-4
- Parallel: true (independent of graders, only needs persistence)

## Definition of Done
- [ ] Generator script implemented
- [ ] At least 1 sample fixture committed
- [ ] Parametrized test suite runs
- [ ] `regression` marker registered
- [ ] Tests pass with TestModel
- [ ] No lint/type errors
